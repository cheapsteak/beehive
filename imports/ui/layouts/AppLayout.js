// Framework
import React from "react";
import { Switch, Route, Link } from "react-router-dom";

// Components
import { Container, Row, Col, Button } from "reactstrap";
import Navigation from "../components/Navigation";

import HomeContainer from "../containers/HomeContainer";
import ConfigContainer from "../containers/ConfigContainer";
import SettingsContainer from "../containers/SettingsContainer";
import NotFound from "../components/NotFound";

const navigationItems = [
  { label: "Game", path: "/" },
  { label: "Settings", path: "/settings" },
  { label: "Config", path: "/config" }
];

/**
 * Main rendering entry point for the app.
 *
 * @returns {Element} Stateless functional React component.
 */
const AppLayout = ({ children }) => (
  <Container className="app px-0" fluid>
    <Row className="bg-faded">
      <Col>
        <header className="py-4 text-center">
          <h1>Beehive Simulator</h1>
        </header>
      </Col>
    </Row>
    <Row className="bg-faded">
      <Col>
        <Navigation navigationItems={navigationItems} />
      </Col>
    </Row>
    <Row noGutters>
      <Col>
        <main>
          <Switch>
            <Route exact path="/" component={HomeContainer} />
            <Route exact path="/config" component={ConfigContainer} />
            <Route exact path="/settings" component={SettingsContainer} />
            <Route path="*" component={NotFound} />
          </Switch>
        </main>
      </Col>
    </Row>
    <Row>
      <Col>
        <footer className="py-1 text-center">
          <p>
            Created by: <a href="https://mackbrowne.com/">Mack Browne</a> and{" "}
            <a href="https://mackiedrew.com">Mackie Drew</a>
          </p>
        </footer>
      </Col>
    </Row>
  </Container>
);

export default AppLayout;
