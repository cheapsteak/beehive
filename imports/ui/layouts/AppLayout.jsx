// Framework
import React from "react";
import { Switch, Route, Link } from "react-router-dom";

// Components
import { Container, Row, Col, Button } from "reactstrap";
import Navigation from "../components/Navigation";

import HomeContainer from "../containers/HomeContainer";
import ConfigContainer from "../containers/ConfigContainer";
import TimesheetsContainer from "../containers/TimesheetsContainer";
import NotFound from "../components/NotFound";

const navigationItems = [
  { label: "Report", path: "/" },
  { label: "Timesheets", path: "/timesheets" },
  { label: "Config", path: "/config" }
];

/**
 * Main rendering entry point for the app.
 *
 * @returns {Element} Stateless functional React component.
 */
const AppLayout = ({ children }) =>
  <Container className="app px-0" fluid>
    <Row className="bg-faded">
      <Col>
        <header className="py-4 text-center">
          <h1>Wave Payroll Tool</h1>
        </header>
      </Col>
    </Row>
    <Row className="bg-faded">
      <Col>
        <Navigation navigationItems={navigationItems} />
      </Col>
    </Row>
    <Row noGutters>
      <Col>
        <main>
          <Switch>
            <Route exact path="/" component={HomeContainer} />
            <Route exact path="/config" component={ConfigContainer} />
            <Route exact path="/timesheets" component={TimesheetsContainer} />
            <Route path="*" component={NotFound} />
          </Switch>
        </main>
      </Col>
    </Row>
    <Row>
      <Col>
        <footer className="py-1 text-center">
          <p>
            Created by: <a href="https://mackbrowne.com/">Mack Browne</a>
          </p>
        </footer>
      </Col>
    </Row>
  </Container>;

export default AppLayout;
