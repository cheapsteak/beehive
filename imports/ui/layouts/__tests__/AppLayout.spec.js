import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import ReactTestRenderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import AppLayout from '../AppLayout';

describe('<AppLayout />', () => {
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      children: []
    };
  });

  describe('AppLayout SnapShot', () => {
    it('renders config correctly with default props', () => {
      delete defaultProps.configFields;
      const tree = ReactTestRenderer.create(
        <MemoryRouter initialEntries={['/home']}>
          <AppLayout {...defaultProps} />
        </MemoryRouter>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  // describe("AppLayout Props", () => {
  //   let wrapper, fieldRows;
  //
  //   beforeEach(() => {
  //     const wrapper = shallow(
  //       <AppLayout {...defaultProps} />
  //     );
  //     fieldRows = wrapper.find(FieldRows);
  //   });
  //
  //   it("should pass fields prop to FieldRows", () => {
  //     const { fields } = fieldRows.props();
  //     expect(fields).toBe(defaultProps.configFields);
  //   });
  // });
});
