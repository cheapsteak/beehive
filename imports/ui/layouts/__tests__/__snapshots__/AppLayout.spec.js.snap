// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppLayout /> AppLayout SnapShot renders config correctly with default props 1`] = `
<div
  className="app px-0 container-fluid"
>
  <div
    className="bg-faded row"
  >
    <div
      className="col"
    >
      <header
        className="py-4 text-center"
      >
        <h1>
          Wave Payroll Tool
        </h1>
      </header>
    </div>
  </div>
  <div
    className="bg-faded row"
  >
    <div
      className="col"
    >
      <nav
        className="navigation"
      >
        <div
          className="container-fluid"
        >
          <div
            className="row"
          >
            <div
              className="text-center p-1 col"
            >
              <a
                className="navigation-item"
                href="/"
                onClick={[Function]}
              >
                Report
              </a>
            </div>
            <div
              className="text-center p-1 col"
            >
              <a
                className="navigation-item"
                href="/timesheets"
                onClick={[Function]}
              >
                Timesheets
              </a>
            </div>
            <div
              className="text-center p-1 col"
            >
              <a
                className="navigation-item"
                href="/config"
                onClick={[Function]}
              >
                Config
              </a>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </div>
  <div
    className="no-gutters row"
  >
    <div
      className="col"
    >
      <main>
        <div
          className="not-found container"
        >
          <div
            className="row"
          >
            <div
              className="col"
            >
              <p>
                <strong>
                  Error [404]
                </strong>
                : 
                blank
                 does not exist.
              </p>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <footer
        className="py-1 text-center"
      >
        <p>
          Created by: 
          <a
            href="https://mackbrowne.com/"
          >
            Mack Browne
          </a>
<<<<<<< HEAD
=======
           and
           
          <a
            href="https://mackiedrew.com"
          >
            Mackie Drew
          </a>
>>>>>>> master
        </p>
      </footer>
    </div>
  </div>
</div>
`;
